main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,152 @@
+}
+
+const MethodImplementationResults findMethodImplementations(const core::GlobalState &gs, core::SymbolRef method) {
+    if (!method.data(gs)->isMethod() || !method.data(gs)->isAbstract()) {
+        return {.error = {makeInvalidParamsError(
+                    "Go to implementation can be used only for methods or references of abstract classes")}};
+    }
+
+    vector<core::Loc> locations;
+    auto owner = method.data(gs)->owner;
+    if (!owner.isClassOrModule())
+        return {.error = {makeInvalidParamsError("Abstract method can only be inside a class or module")}};
+
+    auto owningClassSymbolRef = owner.asClassOrModuleRef();
+    auto childClasses = owningClassSymbolRef.getSubclasses(gs, false);
+    auto methodName = method.data(gs)->name;
+    for (const auto &childClass : childClasses) {
+        auto methodImplementation = childClass.data(gs)->findMember(gs, methodName);
+        locations.push_back(methodImplementation.data(gs)->loc());
+    }
+
+    return {.locations = {locations}};
+}
+
+core::SymbolRef findOverridedMethod(const core::GlobalState &gs, const core::SymbolRef method) {
+    auto ownerClass = method.data(gs)->owner.asClassOrModuleRef();
+
+    for (auto mixin : ownerClass.data(gs)->mixins()) {
+        if (!mixin.data(gs)->isClassOrModule() && !mixin.data(gs)->isAbstract()) {
+            continue;
+        }
+        return mixin.data(gs)->findMember(gs, method.data(gs)->name);
+    }
+    return core::Symbols::noSymbol();
+}
+
+GoToImplementationTask::GoToImplementationTask(const LSPConfiguration &config, MessageId id,
+                                               std::unique_ptr<ImplementationParams> params)
+    : LSPRequestTask(config, move(id), LSPMethod::TextDocumentImplementation), params(move(params)) {}
+
+unique_ptr<ResponseMessage> GoToImplementationTask::runRequest(LSPTypecheckerDelegate &typechecker) {
+    auto response = make_unique<ResponseMessage>("2.0", id, LSPMethod::TextDocumentImplementation);
+
+    const core::GlobalState &gs = typechecker.state();
+    auto queryResult =
+        queryByLoc(typechecker, params->textDocument->uri, *params->position, LSPMethod::TextDocumentImplementation);
+
+    if (queryResult.error) {
+        // An error happened while setting up the query.
+        response->error = move(queryResult.error);
+        return response;
+    }
+
+    if (queryResult.responses.empty()) {
+        return response;
+    }
+
+    vector<unique_ptr<Location>> result;
+    auto queryResponse = move(queryResult.responses[0]);
+    if (auto def = queryResponse->isDefinition()) {
+        // User called "Go to Implementation" from the abstract function definition
+        core::SymbolRef method = def->symbol;
+        if (!method.data(gs)->isMethod())
+            response->error = make_unique<ResponseError>(
+                (int)LSPErrorCodes::InvalidParams,
+                "Go to implementation can be used only for methods or references of abstract classes");
```

Comment:
Could you put braces around this conditional? Also, does it make sense to continue processing results at this point? The body of the conditionals on line 106 and 127 return after setting the error code, which makes me think it might be worth doing the same here.